packages/backend/src/core/SequenceProcessor.ts

```diff
@@ -117,9 +121,14 @@ export class SequenceProcessor extends EventEmitter {
   private async process(): Promise<void> {
```

Comment:
This function is pretty big and hard to read without greater context. Definietely some comment at the top would be useful, or to be more clean splitting it into smaller pieces to be more self-explainatory. But this one should not be in the scope of this PR.